docs/pages/tutorial/layout.md

```diff
@@ -0,0 +1,467 @@
+    alignItems: 'center',
+  },
+});
+/* @end */
+```
+
+## Step 6: Create buttons using Pressable
+
+A modern mobile device typically constitutes a touch screen as the primary interface between an application and the user. Any level of event interaction in an application can be achieved by using a button that, on a physical device, is a simple finger touch.
+
+React Native provides various components to handle touch events on native platforms. For this tutorial, let's use the [`Pressable`](https://reactnative.dev/docs/pressable). It is a core component wrapper that can detect various stages of interactions, from basic single tap events to advance events such as the long press. It can be customized in terms of style and appearance, and you can extend the functionality by creating a custom wrapper component around it.
+
+From the layout, there are two buttons you have to implement. Each has different styles and labels. Let's start by creating a function component that can be reused to create the two buttons.
+
+- Create a new file called **Button.js** inside the **components** folder.
+- Then, import the `Pressable` component from React Native
+- Create a custom component called `Button` with the following code snippet:
+
+<!-- prettier-ignore -->
+```js
+import { StyleSheet, View, Pressable, Text } from 'react-native';
+
+export default function Button({ label }) {
+  return (
+    <View style={styles.buttonContainer}>
+      <Pressable style={styles.button} onPress={() => alert('You pressed a button.')}>
+        <Text style={styles.buttonLabel}>{label}</Text>
+      </Pressable>
+    </View>
+  );
+}
+
+const styles = StyleSheet.create({
+  buttonContainer: {
+    width: 320,
+    height: 68,
+    marginHorizontal: 20,
+    alignItems: 'center',
+    justifyContent: 'center',
+    padding: 3,
+  },
+  button: {
+    borderRadius: 10,
+    width: '100%',
+    height: '100%',
+    alignItems: 'center',
+    justifyContent: 'center',
+    flexDirection: 'row',
+  },
+  buttonIcon: {
+    paddingRight: 8,
+  },
+  buttonLabel: {
+    color: '#fff',
+    fontSize: 16,
+  },
+});
+```
+
+The `Button` component receives a `label` [prop](https://reactnative.dev/docs/intro-react#props) so that anytime this component is used, the button can have a text displayed.
+
+Now, in the app, when the user will touch any of the buttons on the screen, a dialog box will be displayed. This is done by using an `alert()` function to display a static message. Currently, it is used as a placeholder since both buttons do not trigger any functionality right now.
+
+Now let's import this component into **App.js** and see it in action:
+
+<!-- prettier-ignore -->
+```js
+import { StatusBar } from "expo-status-bar";
+import { StyleSheet, View} from "react-native";
+
+/* @info */ import Button from './components/Button'; /* @end */
+
+import ImageViewer from './components/ImageViewer';
+
+const PlaceholderImage = require("./assets/images/background-image.png");
```

Comment:
I usually follow the require pattern when importing images and passing them dynamically.

Later when the ImageViewer component has two different types of sources of image:
- one is a local image
- one is picked from image picker as an `uri`

Instead of writing two Image components, I'm passing both of these props dynamically. The static resource uses the syntax `source` and `require` where as the image picked from the Image Picker library will need `uri`.

In the Image Picker chapter, you will find it the following code snippet:

```js
<Image
        source={selectedImage !== null ? { uri: selectedImage } : placeholderImageSource}
        style={styles.image}
      />
```

I don't know if there is a better way to handle this.